hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/HandleContext.java

```diff
@@ -119,6 +119,13 @@ enum PrecedingTransactionCategory {
     @NonNull
     AccountID payer();
 
+    /**
+     * Attempts to charge the payer in this context the given amount.
+     * @param amount the amount to charge
+     * @throws IllegalArgumentException if the payer cannot afford the fee
```

Comment:
`IllegalArgumentException` is sort of a stretch, that's usually for "static" constraints, like, must be positive.  Where this is "dynamic", if you see what I mean.  Sticking with JDK types, perhaps `RangeException`?

Actually a better option would be to have this method return something.  Even just a `boolean`.  Skip the exception (which already is not a pattern used in this `HandleContext` class except for a couple of places).

Reply:
Yes I forgot to mark the initial PR "Draft"...was veryy half-baked. 😳 Changed this.